framework/vectorstore/redis.go

```diff
@@ -0,0 +1,880 @@
+	}
+
+	// Create key with namespace
+	key := buildKey(namespace, id)
+
+	// Prepare hash fields: binary embedding + metadata
+	fields := make(map[string]interface{})
+
+	// Only add embedding if it's not empty
+	if len(embedding) > 0 {
+		// Convert float32 slice to bytes for Redis storage
+		embeddingBytes := float32SliceToBytes(embedding)
+		fields["embedding"] = embeddingBytes
+	}
+
+	// Add metadata fields directly (no prefix needed with proper indexing)
+	for k, v := range metadata {
+		switch val := v.(type) {
+		case string:
+			fields[k] = val
+		case int, int64, float64, bool:
+			fields[k] = fmt.Sprintf("%v", val)
+		case []interface{}:
+			// Preserve arrays as JSON to support round-trips (e.g., stream_chunks)
+			b, err := json.Marshal(val)
+			if err != nil {
+				return fmt.Errorf("failed to marshal array metadata %s: %w", k, err)
+			}
+			fields[k] = string(b)
+		default:
+			// JSON encode complex types
+			jsonData, err := json.Marshal(val)
+			if err != nil {
+				return fmt.Errorf("failed to marshal metadata field %s: %w", k, err)
+			}
+			fields[k] = string(jsonData)
+		}
+	}
+
+	// Store as hash for efficient native vector search
+	if err := s.client.HSet(ctx, key, fields).Err(); err != nil {
+		return fmt.Errorf("failed to store semantic cache entry: %w", err)
+	}
+
+	return nil
+}
+
+func (s *RedisStore) Delete(ctx context.Context, namespace string, id string) error {
+	ctx, cancel := withTimeout(ctx, s.config.ContextTimeout)
+	defer cancel()
+
+	if strings.TrimSpace(id) == "" {
+		return fmt.Errorf("id is required")
+	}
+
+	// Create key with namespace
+	key := buildKey(namespace, id)
+
+	// Delete the hash key
+	result := s.client.Del(ctx, key)
+	if result.Err() != nil {
+		return fmt.Errorf("failed to delete chunk %s: %w", id, result.Err())
+	}
+
+	// Check if the key actually existed
+	if result.Val() == 0 {
+		return fmt.Errorf("chunk not found: %s", id)
+	}
+
+	return nil
+}
+
+func (s *RedisStore) DeleteAll(ctx context.Context, namespace string, queries []Query) ([]DeleteResult, error) {
+	ctx, cancel := withTimeout(ctx, s.config.ContextTimeout)
+	defer cancel()
+
+	// Use cursor-based deletion to handle large datasets efficiently
+	return s.deleteAllWithCursor(ctx, namespace, queries, nil)
+}
+
+// deleteAllWithCursor performs cursor-based deletion for large datasets
+func (s *RedisStore) deleteAllWithCursor(ctx context.Context, namespace string, queries []Query, cursor *string) ([]DeleteResult, error) {
+	// Get a batch of documents to delete (using pagination)
+	results, nextCursor, err := s.GetAll(ctx, namespace, queries, []string{}, cursor, BatchLimit)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find documents to delete: %w", err)
+	}
+
+	if len(results) == 0 {
+		return []DeleteResult{}, nil
+	}
+
+	// Extract IDs from results
+	ids := make([]string, len(results))
+	for i, result := range results {
+		ids[i] = result.ID
+	}
+
+	// Delete this batch of documents
+	var deleteResults []DeleteResult
```

Comment:
you can use the defaultBatch size define above, lets not have random const defined in between